security/ldap-config-handler/src/main/java/org/codice/ddf/admin/security/ldap/test/LdapTestingCommons.java

```diff
@@ -194,6 +199,55 @@ private static BindRequest selectBindMethod(String bindMethod, String bindUser,
         return request;
     }
 
+    /**
+     * Checks for existing LDAP configurations with the same hostname and port of the {@code configuration}.
+     * If there is an existing configuration, warnings will be returned. If there are no existing configurations
+     * an empty {@link List} will be returned.
+     *
+     * @param configuration configuration to check for existing configurations for
+     * @return {@link List} with {@link ConfigurationMessage}s containing warnings if there are existing configurations that match
+     * the {@code configuration}, or empty {@link List} if no matches
+     */
+    public static List<ConfigurationMessage> ldapConnectionExists(LdapConfiguration configuration) {
+        LdapConfigurationHandler ldapConfigurationHandler = new LdapConfigurationHandler();
+        List<LdapConfiguration> ldapConfigs = ldapConfigurationHandler.getConfigurations();
+
+        String ldapUseCase = configuration.ldapUseCase();
+        String msg = null;
+        switch (ldapUseCase) {
+        case LdapValidationUtils.ATTRIBUTE_STORE:
+            msg = "LDAP Attribute Store";
+        case LdapValidationUtils.AUTHENTICATION:
+            if (msg == null) {
+                msg = "LDAP Authentication Source";
+            }
+
+            for (LdapConfiguration ldapConfig : ldapConfigs) {
+                if (ldapConfig.ldapUseCase()
+                        .equals(ldapUseCase) && configurationExists(ldapConfig, configuration)) {
+                    return createDuplicationWarning(msg, ldapConfig.hostName(), ldapConfig.port());
+                }
+            }
+            break;
+        case LdapValidationUtils.AUTHENTICATION_AND_ATTRIBUTE_STORE:
+            for (LdapConfiguration ldapConfig : ldapConfigs) {
+                if (configurationExists(ldapConfig, configuration)) {
+                    return createDuplicationWarning(msg, ldapConfig.hostName(), ldapConfig.port());
+                }
+            }
+            break;
+        default:
+            LOGGER.debug(
+                    "Failed to validate against existing configurations. Invalid use case \"{}\".",
```

Comment:
❗️ We've been using square brackets and not quotation marks. No need to escape them.